src/logics/Logic.cc

```diff
@@ -82,9 +82,7 @@ Logic::Logic(opensmt::Logic_t _logicType) :
     sortToEquality.insert(sort_BOOL, sym_EQ);
     sortToDisequality.insert(sort_BOOL, sym_DISTINCT);
     sortToIte.insert(sort_BOOL, sym_ITE);
-    if (hasArrays()) {
-        sym_ArraySort = sort_store.newSortSymbol(SortSymbol("Array", 2, SortSymbol::INTERNAL));
-    }
+    sym_ArraySort = hasArrays() ? sort_store.newSortSymbol(SortSymbol("Array", 2, SortSymbol::INTERNAL)) : SSymRef_Undef;
```

Comment:
I would actually prefer to always create this sort properly, even though this suggested fix is more consistent with the philosophy that SMT-LIB logic type dictates which terms are possible to create.